lib/galaxy/tools/__init__.py

```diff
@@ -622,6 +622,21 @@ def parse(self, tool_source, guid=None):
 
         self.parse_command(tool_source)
         self.environment_variables = self.parse_environment_variables(tool_source)
+        self.tmp_directories = tool_source.parse_tmp_directories()
```

Comment:
Is this more accurately `tmp_directory_vars` or `tmp_directory_varnames`?

Reply:
Yes that would clarify this a lot I think.